halo/halo.py

```diff
@@ -94,12 +94,8 @@ def __enter__(self):
         return self.start()
 
     def __exit__(self, type, value, traceback):
-        """Stops the spinner. For use in context managers.
-        Returns
-        -------
-        None
-        """
-        return self.stop()
+        """Stops the spinner. For use in context managers."""
+        self.stop()
```

Comment:
Does this mean, the exception is raised but eaten up by context manager because of `return`?

Reply:
Aye, if [`__exit__`](https://docs.python.org/3/reference/datamodel.html#object.__exit__) returns a truthy value, the exception gets swallowed, and execution continues with the next line (still inside the `with`). `self.stop()` returns `self`, which is an `object`, and those are truthy by default, unless a `__bool__` (or `__nonzero__` in py2, IIRC) method is defined and returns False.

Reply:
Thanks for explaining. I learned something new! 😄